packages/react-admin/src/sideEffect/saga/crudFetch.js

```diff
@@ -18,13 +19,12 @@ const crudFetch = dataProvider => {
         const { type, payload, meta: { fetch: fetchMeta, ...meta } } = action;
         const restType = fetchMeta;
 
-        yield all([
-            put({ type: `${type}_LOADING`, payload, meta }),
-            put({ type: FETCH_START }),
-        ]);
-        let response;
         try {
-            response = yield call(
+            yield all([
+                put({ type: `${type}_LOADING`, payload, meta }),
+                put({ type: FETCH_START }),
+            ]);
+            let response = yield call(
```

Comment:
Why those changes ? They don't seem related to this PR

Reply:
They are, because the generator can be `yielding`  on the puts, while `cancel` is called from the `watchCrudFetch` generator, if this happens the `cancel` finalizer will never run, and the loading state isn't reduced correctly.